### PR TITLE
Metadata and store assets

### PR DESCRIPTION
**Pull request by 6r1d**
Part of https://github.com/hyperledger/iroha-2-docs/issues/308

### CODE REVIEW

**Comment by 6r1d** [comment]
I have a question on the topic of this document. I want to tell what could be helpful for our users in this list of possible values for `Value`, which is key for the `Store` `AssetDefinition` instances.

Those are possible values. Which ones should I add, and which ones can I ignore, besides strings, booleans and numbers?

* Bool
* String
* Name
* LimitedMetadata
* MetadataLimits
* TransactionLimits
* LengthLimits
* Id
* Identifiable
* PublicKey
* SignatureCheckCondition
* TransactionValue
* TransactionQueryResult
* PermissionToken
* Hash
* Block
* BlockHeader
* Ipv4Addr
* Ipv6Addr
* Numeric
* Validator

File: src/guide/configure/metadata-and-store-assets.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Choosing between the Store assets and metadata
+
+The `Store` assets and `metadata` allow storing several parameters with different types and have string keys. Despite the similarity, their use cases differ.
+
+You must use `metadata` when you don’t want sequences of records, for example, extending user data. You’ll extend its `metadata` when you need to store something intended to be added to a specific `Account` or `Domain` instance.
+
+You must use a `Store` asset if the data exists in the blockchain and a default number type is not applicable. It lets you record data sequences and works as a dictionary, where the keys are string, while the values are  `Value` type instance that assumes many types starting with a `String` and finishing with `Ipv6Addr`.
```
Comment (outoftardis) [inline src/guide/configure/metadata-and-store-assets.md:7]:
```suggestion
You must use a `Store` asset if the data exists in the blockchain and a default number type is not applicable. It lets you record data sequences and works as a dictionary, where the keys are strings, while the values are  `Value` type instances that assume many types from `String` to `Ipv6Addr`.
```

Comment (6r1d) [reply src/guide/configure/metadata-and-store-assets.md]: Thanks!

File: src/guide/configure/metadata-and-store-assets.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Choosing between the Store assets and metadata
+
+The `Store` assets and `metadata` allow storing several parameters with different types and have string keys. Despite the similarity, their use cases differ.
```
Comment (WRRicht3r) [inline src/guide/configure/metadata-and-store-assets.md:3]:
```suggestion
The `Store` and `metadata` assets allow for the storage of several parameters with different types and string keys. Despite the similarity, their use cases differ.
```

File: src/guide/configure/metadata-and-store-assets.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+You must use `metadata` when you don’t want sequences of records, for example, extending user data. You’ll extend its `metadata` when you need to store something intended to be added to a specific `Account` or `Domain` instance.
```
Comment (WRRicht3r) [inline src/guide/configure/metadata-and-store-assets.md:5]:
```suggestion
You must use `metadata` when you don’t want sequences of records, for example, extending user data. You can extend its `metadata` when you need to store something intended to be added to a specific `Account` or `Domain` instance.
```

File: src/guide/configure/metadata-and-store-assets.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+You must use a `Store` asset if the data exists in the blockchain and a default number type is not applicable. It lets you record data sequences and works as a dictionary, where the keys are strings, while the values are  `Value` type instances that assume many types from `String` to `Ipv6Addr`.
+
+Let’s discuss some use cases to make it more clear.
+
+Let’s say you have a clinic. Here, generic user data, such as birthdays, can be added to the `metadata`. The `Store` asset instances record the appointments, treatments suggested by the doctors, and medical test results.
```
Comment (WRRicht3r) [inline src/guide/configure/metadata-and-store-assets.md:11]:
```suggestion
Let’s say, for example, you have a clinic. Here, generic user data, such as birthdays, can be added to `metadata`. The `Store` asset instances record the appointments, treatments suggested by the doctors, and medical test results.
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

File: .vitepress/config.ts [hunk 5]
```diff
@@ -213,6 +213,10 @@ function getGuideSidebar(): DefaultTheme.SidebarItem[] {
               text: 'Genesis Block',
               link: '/guide/configure/genesis',
             },
+            {
+              text: 'Metadata and Store assets',
+              link: '/guide/configure/metadata-and-store-assets.md',
```
Comment (0x009922) [inline .vitepress/config.ts:218]:
```suggestion
              link: '/guide/configure/metadata-and-store-assets',
```

**Review by 0x009922 — APPROVED** [review]
Config LGTM

**Review by WRRicht3r — APPROVED** [review]

**Pull request merged**